js/graviex.js

```diff
@@ -0,0 +1,231 @@
+                'public': {
+                    'get': [
+                        'markets',
+                        'tickers',
+                        'trades', // Requires market XXXYYY. Optional: limit, timestamp, from, to, order_by
+                    ],
+                },
+                'private': {
+                },
+            },
+            'fees': {
+                'trading': {
+                    'percentage': true,
+                    'maker': this.parseNumber ('0.001'),
+                    'taker': this.parseNumber ('0.2'),
+                },
+                'funding': {
+                    'withdraw': {
+                        'BTC': 0.0004,
+                        'ETH': 0.022,
+                        'DOGE': 2.0,
+                        'NYC': 1.0,
+                        'XMR': 0.02,
+                        'PIVX': 0.2,
+                        'NEM': 0.05,
+                        'SCAVO': 5.0,
+                        'SEDO': 5.0,
+                        'USDT': 3.0,
+                        'GDM': 0.3,
+                        'PIRL': 0.005,
+                        'PK': 0.1,
+                        'ORM': 10,
+                        'NCP': 10,
+                        'ETM': 10,
+                        'USD': 0,
+                        'EUR': 0,
+                        'RUB': 0,
+                        'other': 0.002,
+                    },
+                },
+            },
+            'limits': {
+                'amount': {
+                    'min': 0.001,
+                    'max': undefined,
+                },
+            },
+            'precision': {
+                'amount': 8,
+                'price': 8,
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const response = await this.publicGetMarkets ();
+        const markets = response;
+        const result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const id = markets[i]['id'];
+            const symbolParts = markets[i]['name'].split ('/');
+            const baseId = symbolParts[0];
+            const quoteId = symbolParts[1];
+            const base = this.commonCurrencyCode (baseId);
+            const quote = this.commonCurrencyCode (quoteId);
+            const symbol = base + '/' + quote;
+            const active = true;
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': active,
+                'info': markets[i],
+            });
+        }
+        return result;
+    }
+
+    parseTicker (ticker, market = undefined) {
+        const timestamp = this.milliseconds ();
+        const symbol = market['symbol'];
```

Comment:
The value of `market` may be `undefined` here, so a lookup of `market['symbol']` inside an undefined market will break the code, as it follows from the arguments, so we have to do the safeMarket approach here, as we do in the other exchanges' parseTicker implementations.

Reply:
Ok. Fixed